air-quality-ui/src/components/single-city/SingleCity.tsx

```diff
@@ -163,10 +163,12 @@ export const SingleCity = () => {
   const [siteColors, setSiteColors] = useState<Record<string, string>>({})
   useEffect(() => {
     const updateColors = async () => {
+      let index = 0
       const colorsBySite: Record<string, string> = {}
       for (const { value } of sites) {
-        const color = await textToColor(value)
+        const color = await indexToColor(index)
         colorsBySite[value] = color
+        index++
       }
```

Comment:
could you not make use of the following? 

sites.foreach((value, index) => {})

Reply:
Yes, much nicer now, thanks!